core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1989,23 +1989,35 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
       newBitChunk(chunk.take(toTake), minBitIndex, index)
     }
 
+    // Iterates over bits in 3 phases:
+    // 1. Prefix bits: individual bits before the first full element
+    // 2. Full elements: process entire bytes/ints/longs efficiently via foreachElement
+    // 3. Suffix bits: individual bits after the last full element
     override def foreach[A](f: Boolean => A): Unit = {
-      val minLongIndex    = (minBitIndex + bits - 1) >> bitsLog2
-      val maxLongIndex    = maxBitIndex >> bitsLog2
-      val minFullBitIndex = (minLongIndex << bitsLog2) min maxBitIndex
-      val maxFullBitIndex = (maxLongIndex << bitsLog2) max minFullBitIndex
-      var i               = minBitIndex
-      while (i < minFullBitIndex) {
+      val minElementIndex = (minBitIndex + bits - 1) >> bitsLog2 // first full element index
+      val maxElementIndex = maxBitIndex >> bitsLog2              // last full element index (exclusive)
+      val minFullBitIndex = (minElementIndex << bitsLog2) min maxBitIndex
+      val maxFullBitIndex = (maxElementIndex << bitsLog2) max minFullBitIndex
+      val prefixBits      = minFullBitIndex - minBitIndex        // count of leading bits before first full element
+      val suffixBitsStart = maxFullBitIndex - minBitIndex        // 0-based index where trailing bits start
```

Comment:
What really worries me here is that we're assuming the calculations are correct and yet we have no guarantees that:
1. `prefixBits == minElementIndex - 1`
2. `maxElementIndex == suffixBitsStart - 1`
3. `prefixBits < length`
4. `suffixBitsStart < length`